### ISSUE TITLE
Build failure: git-branchless

### ISSUE THREAD

**Issue by toastal** [NixOS/nixpkgs#338971]
### Steps To Reproduce

1. build *git-branchless*

### Build log

https://hydra.nixos.org/search?query=git-branchless

### Additional context

No longer builds with latest Nixpkgs

### Notify maintainers

@arxanas

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix run nixpkgs#nix-info -- "-m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.6, NixOS, 24.11 (Vicuna), 24.11.20240828.71e91c4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.92.0-dev-pre20240813-f9a3bf6
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/toastal/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/toastal/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/toastal/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/toastal/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/3g35mlki1l0p3xx68q6gpwpi5xavgrky-lix-2.92.0-dev-pre20240813-f9a3bf6/share`
 - nixpkgs: `/nix/store/887hpp8a2i99n9jjwcvz6qkhhhqsvzkg-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

[^1]

[^1]: Please consider [giving up MS GitHub](https://sfconservancy.org/GiveUpGitHub/) or offering a non-proprietary, non-US-corporate-controlled mirror for this free software project. I wish to delete this Microsoft account in the future, but I need more projects like this to support alternative methods to send patches & contribute.

**Comment by eclairevoyant** [comment]
As mentioned in https://github.com/NixOS/nixpkgs/pull/339023#issuecomment-2328958177:

> > Why are there tests that are not `_pty` or `_interactive` in there? These shouldn't fail.
> 
> I did run a `git bisect` the other day and it started failing with the update of Git in nixos-unstable to 2.46.0. Users seem to be reporting issues with Git 2.46.0 as well, so I'm not sure we should disable those tests indeed:
> 
> * [Could not parse reference-transaction-line arxanas/git-branchless#1388](https://github.com/arxanas/git-branchless/issues/1388)
> 
> * [I have this problem when switching branches arxanas/git-branchless#1381](https://github.com/arxanas/git-branchless/issues/1381)
> 
> 
> Those issues don't seem to have been traced back to Git 2.46.0 yet, but "updates to Git" are mentioned in [arxanas/git-branchless#1321 (comment)](https://github.com/arxanas/git-branchless/issues/1321#issuecomment-2125636129)

**Comment by niklaskorz** [comment]
The incompatibility with Git 2.45 and 2.46 is [already fixed upstream](https://github.com/arxanas/git-branchless/pull/1322), but pending a new release. Maybe the relevant PR could be applied as a patch until the release of 0.10.0.

**Comment by eclairevoyant** [comment]
Thanks for the link, it seems that `test_amend` is still failing though (see https://github.com/arxanas/git-branchless/pull/1322#issuecomment-2132410167 for context)

**Comment by niklaskorz** [comment]
> Thanks for the link, it seems that `test_amend` is still failing though (see [arxanas/git-branchless#1322 (comment)](https://github.com/arxanas/git-branchless/pull/1322#issuecomment-2132410167) for context)

You are unfortunately correct, building the latest `master` branch still gives these failing tests with Git 2.46:

```
       > failures:
       >     test_amend_merge
       >     test_amend_undo
       >     test_amend_undo_detached_head
       >     test_amend_with_children
       >     test_amend_with_working_copy
```

**Comment by eclairevoyant** [comment]
Well for me on linux there's only one test failing, but nevertheless...

```

failures:
    test_amend_undo

test result: FAILED. 16 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.75s
```

**Comment by niklaskorz** [comment]
> Well for me on linux in english locale, there's only one test failing, but nevertheless...
> 
> ```
> 
> failures:
>     test_amend_undo
> 
> test result: FAILED. 16 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.75s
> ```

In [this issue](https://github.com/arxanas/git-branchless/issues/1345), @arxanas states a failing `test_amend_undo` is safe to ignore.

**Comment by bryango** [comment]
I am trying to revive the fix in #342278, please check out the comments there! Thanks!

**Comment by niklaskorz** [comment]
arxanas is preparing the release of 0.10.0: https://github.com/arxanas/git-branchless/pull/1418
